src/test/java/org/broadinstitute/hellbender/utils/clipping/ClippingOpUnitTest.java

```diff
@@ -0,0 +1,63 @@
+package org.broadinstitute.hellbender.utils.clipping;
+
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.TextCigarCodec;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public final class ClippingOpUnitTest extends BaseTest {
+
+    @Test
+    public void testGetNewAlignmentStartOffset() {
+        Cigar c1 = TextCigarCodec.decode("70M");
+        Cigar c1FrontClipped = TextCigarCodec.decode("10S60M");
+        Cigar c1BackClipped = TextCigarCodec.decode("60M10S");
+        Assert.assertEquals(ClippingOp.getNewAlignmentStartOffset(c1FrontClipped, c1),  10, "c1FrontClipped");
+        Assert.assertEquals(ClippingOp.getNewAlignmentStartOffset(c1BackClipped, c1),  0, "c1BackClipped");
+
+        Cigar c2 = TextCigarCodec.decode("30M10N30M");
+        Cigar c2FrontClippedImproper = TextCigarCodec.decode("30S10N30M");
+        Cigar c2FrontClippedPartial = TextCigarCodec.decode("30S5N30M");
+        Cigar c2BackClippedImproper = TextCigarCodec.decode("30M10N30S");
+        Cigar c2FrontClipped = TextCigarCodec.decode("30S30M");
+        Cigar c2BackClipped = TextCigarCodec.decode("30M30S");
+        Cigar c2FrontClippedTrivial = TextCigarCodec.decode("15S15M10N30M");
+        Cigar c2BackClippedTrivial = TextCigarCodec.decode("30M10N15M15S");
+        Assert.assertEquals(ClippingOp.getNewAlignmentStartOffset(c2FrontClippedImproper, c2),  30, "c2FrontClippedImprope");
+        Assert.assertEquals(ClippingOp.getNewAlignmentStartOffset(c2FrontClippedPartial, c2),  35, "c2FrontClippedPartial");
+        Assert.assertEquals(ClippingOp.getNewAlignmentStartOffset(c2BackClippedImproper, c2),  0, "c2BackClippedImproper");
+        Assert.assertEquals(ClippingOp.getNewAlignmentStartOffset(c2FrontClipped, c2),  40, "c2FrontClipped");
+        Assert.assertEquals(ClippingOp.getNewAlignmentStartOffset(c2BackClipped, c2),  0, "c2BackClipped");
+        Assert.assertEquals(ClippingOp.getNewAlignmentStartOffset(c2FrontClippedTrivial, c2),  15, "c2FrontClippedTrivial");
+        Assert.assertEquals(ClippingOp.getNewAlignmentStartOffset(c2BackClippedTrivial, c2),  0, "c2BackClippedTrivial");
```

Comment:
Can you refactor to use a `DataProvider`, with each test case consisting of a `Cigar` plus expected new alignment start offset